config/initializers/content_security_policy.rb

```diff
@@ -2,20 +2,22 @@
 # For further information see the following documentation
 # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy
 
+asset_host = ENV['CDN_HOST'] || ENV['WEB_DOMAIN'] || ENV['LOCAL_DOMAIN']
+
 Rails.application.config.content_security_policy do |p|
+  p.base_uri        :none
+  p.default_src     :none
   p.frame_ancestors :none
-  p.object_src  :none
-  p.script_src  :self
-  p.base_uri :none
-#  p.default_src :self, :https
-#  p.font_src    :self, :https, :data
-#  p.img_src     :self, :https, :data
-#  p.style_src   :self, :https, :unsafe_inline
-#
-#  # Specify URI for violation reports
-#  # p.report_uri "/csp-violation-report-endpoint"
+  p.script_src      :self, "https://#{asset_host}"
+  p.font_src        :self, "https://#{asset_host}"
+  p.img_src         :self, :https, :data, :blob
```

Comment:
I'm not sure we use `:data` and I guess we only load from `:self` and/or assets host, hm… `:blob` is needed for the drawing tool.